### ISSUE TITLE
Issue ingesting blocks with block_height as -1

### ISSUE THREAD

**Issue by garethtdavies** [zcash-hackworks/lightwalletd#35]
After following installation instructions when starting zcashd with -reindex I get the following error:

```
INFO[2019-04-23T19:11:05-07:00] Listening to 0mq on 127.0.0.1:28332           app=zmqclient
INFO[2019-04-23T19:11:38-07:00] new block                                     app=zmqclient block_hash=382c4a332661c7ed0671f32a34d724619f086c61873bce7c99859dd9920aa605block_height=0 block_version=4 error="<nil>" sapling=false seqnum=0 tx_count=1
INFO[2019-04-23T19:11:38-07:00] new block                                     app=zmqclient block_hash=238d665a062b9007836a7d8f968ba2f3847af5f542733389a952cf9b86795502block_height=-1 block_version=4 error="<nil>" sapling=false seqnum=1 tx_count=1
ERROR[2019-04-23T19:11:38-07:00] new block                                     app=zmqclient block_hash=181ac2d6dc0ae73a82d7db472ac2879a7c24d8b12e5f73efc33a55549ea4f100 block_height=-1 block_version=4 error="storing compact block -1: UNIQUE constraint failed: blocks.block_height" sapling=false seqnum=2 tx_count=1
ERROR[2019-04-23T19:11:46-07:00] error on msg recv                             app=zmqclient error="interrupted system call"
```

In addition, viewing the sqlite database for the `blocks` table I see the following for `block_height` 0 and the `compact encoding` field:

```
 8,J3&aq*4$ala;|ْ
 
```

Running go 1.12.2 and zcashd v2.0.4 on Ubuntu 19.0.4.

**Comment by garethtdavies** [comment]
This was a missing dependency. Ensured everything from https://github.com/zcash-hackworks/lightwalletd-z-cash/blob/master/Dockerfiles/lightwalletd/Dockerfile was installed and this cured it.

**Comment by garethtdavies** [comment]
Actually, the above fix was coincidental and the underlying issue remains. 

I have found that for blocks 1-16 this line always returns -1 https://github.com/zcash-hackworks/lightwalletd/blob/master/parser/block.go#L71 so if you ingest a couple of blocks 1-16 then you'll get the `UNIQUE constraint failed` error.

If you don't ingest any blocks 1-16 then everything works as it should.

**Comment by str4d** [comment]
The bug is caused by the fact that the parser doesn't account for how integers 1 through 16 are encoded in a `CScript` using their opcodes instead of a byte representation:

https://github.com/zcash/zcash/blob/8a95ba235d146d58c9a6883673fc5da4b9139d03/src/script/script.h#L365-L368

**Comment by garethtdavies** [comment]
Thanks for the explanation! 

Before this is fixed if anyone is having the same issue simply starting zcashd with -rescan before starting lightwalletd is a _really_ simple workaround as then you'll simply miss those problematic first blocks.

**Comment by mdr0id** [comment]
@garethtdavies I'm not sold on my PR being fully robust or even a valid fix, but FWIW #37 appears to resolve this issue too on my local environment
